src/errno.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
```

Comment:
The nice part about this error type is that it is fully compatible with os error code thrown off by system calls. If one were to put an os error code into an `io::Error`, it might be hard to get the os error back out in upper parts of the call stack. Using `io::Error::raw_os_error` returns an `Option` that either requires `unwrap` or an extra `match` just to extract the code. Using `io::Error::kind` is tricky because it has variants which are not possible to get from a syscall (`UnexpectedEof`), or are mismatched by the std implementation that converts error numbers to `ErrorKind` variants. Additionally, some error codes have no correspondence with an `ErrorKind` variant other than `Other`.

If human error messaging is all that is needed to make this module acceptable, I suggest using what we have in crosvm today for `Error`:

```
impl Display for Error {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        io::Error::from_raw_os_error(self.0).fmt(f)
    }
}
```